src-tauri/src/capturer/mod.rs

```diff
@@ -80,98 +65,108 @@ pub async fn start_capture(app_handle: AppHandle) {
     // record multiple gifs without restarting the app
     *frames = Vec::new();
 
+    // Create file in temp directory
+    let preview_file = format!("HM-{}.mp4", get_random_id());
+    let mut preview_path = state.preview_path.lock().await;
+    *preview_path = Some(
+        NamedTempFile::new()
+            .unwrap()
+            .into_temp_path()
+            .with_file_name(&preview_file),
+    );
+
+    let mut recorder = state.recorder.lock().await;
+    let [output_width, output_height] = (*recorder).as_mut().unwrap().get_output_frame_size();
+    drop(recorder);
+
+    let file_output = encoder::FileOutput {
+        output_filename: preview_path.as_ref().unwrap().to_str().unwrap().to_string(),
+    };
+
+    let mut i = 0;
+
+    let (tx, rx) = mpsc::channel();
+
+    // Spawn a processing thread
+
+    let process_thread = thread::spawn(move || {
+        process_data(
+            rx,
+            output_width as usize,
+            output_height as usize,
+            file_output,
+        );
+    });
+
     loop {
-        let status = state.status.lock().await;
-        if *status != Status::Recording {
+        let mut recorder = state.recorder.lock().await;
+
+        if recorder.is_none() {
+            println!("Exiting encoding loop");
             break;
         }
-        drop(status);
 
-        let mut recorder = state.recorder.lock().await;
         let frame = (*recorder)
             .as_mut()
             .unwrap()
             .get_next_frame()
             .expect("Error");
+
+        let frame_for_preview = frame.clone();
+        tx.send(frame_for_preview).unwrap();
+
         (*frames).push(frame);
-        println!("Frame captured");
+
+        println!("Frame captured {}", i);
+        i += 1;
         drop(recorder);
     }
 
-    println!("Recording stopped");
-}
-
-#[tauri::command]
-pub async fn stop_capture(app_handle: AppHandle) {
-    // Update app state to editing
-    let state = app_handle.state::<AppState>();
-    let mut status = state.status.lock().await;
-    *status = Status::Editing;
-    drop(status);
-
-    // Create file in temp directory
-    let preview_file = format!("HM-{}.mp4", get_random_id());
-    let mut preview_path = state.preview_path.lock().await;
-    *preview_path = Some(
-        NamedTempFile::new()
-            .unwrap()
-            .into_temp_path()
-            .with_file_name(&preview_file),
-    );
+    drop(tx);
+    process_thread.join().expect("Processing thread panicked.");
 
+    println!("Creating Editor Window");
     println!("Preview path: {:?}", preview_path);
+    println!("Preview dimensions: {}x{}", output_width, output_height);
 
-    // Hide cropper, create editor
-    crate::cropper::toggle_cropper(&app_handle);
-    crate::toolbar::toggle_toolbar(&app_handle);
     crate::editor::init_editor(
```

Comment:
At this moment in time when `init_editor()` is called, is the encoded video guaranteed to be ready? The code structure seems to suggest so.

Previously this was not true. Therefore, I fired a `preview-ready` event later (after creating editor window) when the encoding finished. Then the frontend knows that preview video is ready to load.

If the preview is guaranteed to be ready at the time of editor creation now, we can remove this event from both, the backend and frontend.

Reply:
I have added the wait to make sure encoding is done.
```   
 process_thread.join().expect("Processing thread panicked.");
```